migrations/truffle/1_2_witnet_requests.js

```diff
@@ -33,56 +32,73 @@ async function deployWitnetRequests (from, isDryRun, ecosystem, network, request
+            if (
+              utils.isNullAddress(templateAddr) ||
+                (await web3.eth.getCode(templateAddr)).length <= 3
+            ) {
+              const templateArtifact = utils.findTemplateArtifact(witnet.templates, request?.template)
+              if (!templateArtifact) {
+                throw `artifact '${request?.template}' not found in templates file`
+              }
+              templateAddr = await utils.buildWitnetRequestTemplate(
+                web3,
+                from, 
+                request?.template, 
+                templateArtifact, 
+                await WitnetBytecodes.deployed(),
+                await WitnetRequestFactory.deployed(),
+                witnet?.radons,
+                hashes,
+              )
+              if (utils.isNullAddress(templateAddr)) {
+                throw `artifact '${request?.template}' could not get deployed`
+              } else {
+                const templateContract = await WitnetRequestTemplate.at(templateAddr)
+                console.info("  ", `> Template address:  \x1b[1;37m${templateContract.address}\x1b[0m]`)
+                console.info("  ", "> Template registry:", await templateContract.registry.call())
               }
-              utils.traceHeader(`Settling '${key}'...`)
-              console.info("  ", "> Template artifact:", request?.template)
-              console.info("  ", "> Template address: ", templateAddr)
-              const contract = await WitnetRequestTemplate.at(templateAddr)
-              const args = request?.args
-              args.map((subargs, source) => {
-                if (!Array.isArray(subargs)) {
-                  args[source] = Object.values(subargs)
-                }
-                console.info("  ", `> Template source #${source + 1} params => ${JSON.stringify(args[source])}`)
-                return subargs
-              })
-              requestAddress = await utils.buildWitnetRequestFromTemplate(from, contract, request?.args)
-            } catch (ex) {
-              utils.traceHeader(`Failed '${key}': ${ex}`)
-              process.exit(1)
+              addresses[ecosystem][network].templates[request?.template] = templateAddr
             }
+            utils.traceHeader(`Settling '\x1b[1;37m${key}\x1b[0m'...`)
+            console.info("  ", "> Template artifact:", request?.template)
+            console.info("  ", `> Template address:  \x1b[1;37m${templateAddr}\x1b[0m`)
+            const contract = await WitnetRequestTemplate.at(templateAddr)
+            const args = request?.args
+            args.map((subargs, source) => {
+              if (!Array.isArray(subargs)) {
+                args[source] = Object.values(subargs)
+              }
+              console.info("  ", `> Template source #${source + 1} params => \x1b[32m${JSON.stringify(args[source])}\x1b[0m`)
+              return subargs
+            })
```

Comment:
`Array.map`actually collects, clones and returns whatever you return in the mapping function. Because we are using mutability here (`args[source] = ...`), `Array.forEach` seems to be a better fit.

```suggestion
            const args = request?.args
            args.forEach((subargs, source) => {
              if (!Array.isArray(subargs)) {
                args[source] = Object.values(subargs)
              }
              console.info("  ", `> Template source #${source + 1} params => \x1b[32m${JSON.stringify(args[source])}\x1b[0m`)
              return subargs
            })
```

That's certainly the most efficient solution, but there's also a more functional-ish one that gets rid of mutability altogether:

```suggestion
            const args = request?.args.map((subargs, source) => {
              let subargs_array = subargs;
              if (!Array.isArray(subargs)) {
                subargs_array = Object.values(subargs)
              }
              console.info("  ", `> Template source #${source + 1} params => \x1b[32m${JSON.stringify(subargs_array)}\x1b[0m`)
              return subargs_array
            })
```

Reply:
Nice, but if `request?.args` is not mutated, then `args` instead must be passed to web3 call. So I guess the simplest approach would rather be:
```javascript
            const args = request?.args.map((subargs, source) => {
              if (!Array.isArray(subargs)) {
                subargs = Object.values(subargs)
              }
              console.info("  ", `> Template source #${source + 1} params => \x1b[32m${JSON.stringify(subargs)}\x1b[0m`)
              return subargs
            })
            requestAddress = await utils.buildWitnetRequestFromTemplate(web3, from, contract, args)
```